helper/resource/testing_new.go

```diff
@@ -120,6 +120,9 @@ func runNewTest(ctx context.Context, t testing.T, c TestCase, helper *plugintest
 	// acts as default for import tests
 	var appliedCfg string
 
+	// stdout (io.Writer) is supplied to all terraform commands that are using streaming json output.
+	stdout := NewStdout()
```

Comment:
I think we should have each command use its own buffer for a few reasons:

- Prevents the need for everywhere to pass it and for the business logic to deal with any buffering aspects
- Prevents any strange cross-command contamination
- Allows the buffer to be garbage collected, rather than potentially keeping a buffer around that's the largest size from any of the commands it was used in

Reply:
Have made a first pass at this. Interested to hear your thoughts.

I've also left a couple of "TODO"s. Would value your input on these.